openforcefield/tests/test_molecule.py

```diff
@@ -529,6 +529,109 @@ def test_name(self):
         molecule.name = name
         assert molecule.name == name
 
+    def test_hill_formula(self):
+        """Test that making the hill formula is consistent between input methods and ordering"""
+        # make sure smiles match reference
+        molecule_smiles = Molecule.from_smiles('CCO')
+        assert molecule_smiles.hill_formula == 'C2H6O1'
+        # make sure is not order dependent
+        molecule_smiles_reverse = Molecule.from_smiles('OCC')
+        assert molecule_smiles.hill_formula == molecule_smiles_reverse.hill_formula
+        # make sure files and smiles match
+        molecule_file = Molecule.from_file(get_data_file_path('molecules/ethanol.mol2'))
+        assert molecule_smiles.hill_formula == molecule_file.hill_formula
+        # make sure the topology molecule gives the same formula
+        from openforcefield.topology.topology import TopologyMolecule, Topology
+        topology = Topology.from_molecules(molecule_smiles)
+        topmol = TopologyMolecule(molecule_smiles, topology)
+        assert molecule_smiles.hill_formula == Molecule.to_hill_formula(topmol)
+        # make sure the networkx matches
+        assert molecule_smiles.hill_formula == Molecule.to_hill_formula(molecule_smiles.to_networkx())
+
+    def test_isomorphic(self):
+        """Test the different levels of isomorphic matching"""
+        # check that hill formula fails are caught
+        ethanol = Molecule.from_smiles('CCO')
+        acetaldehyde = Molecule.from_smiles('CC=O')
+        assert ethanol.is_isomorphic_with(acetaldehyde) is False
+        assert acetaldehyde.is_isomorphic_with(ethanol) is False
+        # check that different orderings work with full matching
+        ethanol_reverse = Molecule.from_smiles('OCC')
```

Comment:
Same as my comment above -- I will be merging `create_reversed_ethanol` from the branch I linked in my comment above soon. You might consider just copying the function from my branch into yours, and then whoever closes their PR last can resolve the conflict from the duplication. (I'm sure there's a "proper git" way to do this, but given my history will this sort of thing, I'd just break the repo again)